libraries/freertos_plus/standard/pkcs11/include/iot_pkcs11.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Amazon FreeRTOS
+ * Copyright (C) 2017 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+
+#ifndef _AWS_PKCS11_H_
+#define _AWS_PKCS11_H_
+
+#include <stdint.h>
+
+/**
+ * @brief Amazon FreeRTOS PKCS#11 Interface.
+ * The following definitions are required by the PKCS#11 standard public
+ * headers.
+ */
+
+#define CK_PTR          *
+
+#ifndef NULL_PTR
+    #define NULL_PTR    0
+#endif
+
+#define CK_DEFINE_FUNCTION( returnType, name )             returnType name
+#define CK_DECLARE_FUNCTION( returnType, name )            returnType name
+#define CK_DECLARE_FUNCTION_POINTER( returnType, name )    returnType( CK_PTR name )
+#define CK_CALLBACK_FUNCTION( returnType, name )           returnType( CK_PTR name )
+
+/**
+ *   @brief Length of a SHA256 digest, in bytes.
+ */
+#define pkcs11SHA256_DIGEST_LENGTH           32
+
+/**
+ * @brief Length of a curve P-256 ECDSA signature, in bytes.
+ */
+#define pkcs11ECDSA_P256_SIGNATURE_LENGTH    64
```

Comment:
Length of just the non-encoded point values, of course. But do we actually use non-encoded points anywhere?